component/main.jsonnet

```diff
@@ -69,38 +69,43 @@ local alert_rules = import 'alertrules.libsonnet';
         params.clusterLogForwarding.forwarders[fw] { name: fw }
         for fw in std.objectFields(params.clusterLogForwarding.forwarders)
       ],
-      [if std.length(params.clusterLogForwarding.namespaces) > 0 then 'inputs']: [
+      [if std.length(params.clusterLogForwarding.namespace_groups) > 0 then 'inputs']: [
         {
-          name: ns,
+          name: group,
           application: {
-            namespaces: [ ns ],
+            namespaces: params.clusterLogForwarding.namespace_groups[group].namespaces,
           },
         }
-        for ns in std.objectFields(params.clusterLogForwarding.namespaces)
+        for group in std.objectFields(params.clusterLogForwarding.namespace_groups)
       ],
-      [if std.length(params.clusterLogForwarding.namespaces) > 0 then 'pipelines']: [
+      [if std.length(params.clusterLogForwarding.namespace_groups) > 0 then 'pipelines']: [
+        local enable_json = com.getValueOrDefault(params.clusterLogForwarding.namespace_groups[group], 'json', false);
```

Comment:
This PR seems to make two logically distinct changes:

1. Refactor the config to expose namespace groups instead of namespaces for log forwarding
1. Add support for enabling JSON log parsing for custom forwarding

To have a nice changelog, it would make sense to separate those two changes into their own PRs. Additionally, having such changes in separate PRs would make reviewing the changes much easier.

Reply:
true, but these changes are very intertwined. however i removed the json part and do a seperate pr